src/oic/oic/__init__.py

```diff
@@ -258,21 +258,41 @@ def add_token(self, resp):
 }
 
 
-def claims_match(claim1, claim2):
+def claims_match(value, claimspec):
     """
-    Is claim1 equal to or present in claim2?
-
-    :param claim1: First claim value
-    :param claim2: Second claim value
+    Implements matching according to section 5.5.1 of 
+    http://openid.net/specs/openid-connect-core-1_0.html 
+    The lack of value is not checked here.
+    Also the text doesn't prohibit having both 'value' and 'values'.
+
+    :param value: single value or list of values 
+    :param claimspec: None or dictionary with 'essential', 'value' or 'values'
+    as key
     :return: Boolean
     """
-    if claim2 is None:
+    if claimspec is None:  # match anything
         return True
-    if isinstance(claim2, dict):
-        return claim1 in claim2.values()
-    if isinstance(claim2, Iterable):
-        return claim1 in claim2
-    return claim1 == claim2
+
+    matched = False
+    for key, val in claimspec.items():
+        if key == "value":
+            if value == val:
+                matched = True
+        elif key == "values":
+            if value in val:
+                matched = True
+        elif key == 'essential':
```

Comment:
No, if we omit it then the {'essential': True/False} case will fail.
{'essential': True/False} in essence is the same as value == *. 
That is, as long as there is a value it matched.
This means that 'essential' together with 'value'/'values' doesn't add anything. It's only if it's by itself that it matters.